### PR TITLE
Commands: Ignore non-loadable types

### LINKED ISSUE #4683: Cannot generate migration in UWP when using Win2D
I have a UWP project that uses Win2D. When I attempt to generate a migration with EF the migration will fail with the following error.

> System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeAssembly.get_DefinedTypes()
   at Microsoft.Data.Entity.Design.Internal.StartupInvoker..ctor(String startupAssemblyName, String environment)
   at Microsoft.Data.Entity.Design.DbContextOperations..ctor(ILoggerProvider loggerProvider, String assemblyName, String startupAssemblyName, String environment)
   at Microsoft.Data.Entity.Design.MigrationsOperations..ctor(ILoggerProvider loggerProvider, String assemblyName, String startupAssemblyName, String environment, String projectDir, String rootNamespace)
   at Microsoft.Data.Entity.Design.OperationExecutor.<>c__DisplayClass3_0.<.ctor>b__3()
   at Microsoft.Data.Entity.Internal.LazyRef`1.get_Value()
   at Microsoft.Data.Entity.Design.OperationExecutor.<AddMigrationImpl>d__7.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.Data.Entity.Design.OperationExecutor.OperationBase.<>c__DisplayClass4_0`1.<Execute>b__0()
   at Microsoft.Data.Entity.Design.OperationExecutor.OperationBase.Execute(Action action)
Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.

This is due to the use of CanvasBitmap.LoadAsync() from Win2D. Any use of that method will prevent migrations from being generated. I have created a sample project demoing this. You can [find it here](https://github.com/geoffmmorgan/MigrationTest/tree/master)

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #4683

### CODE REVIEW

File: src/Shared/SharedTypeExtensions.cs [hunk 1]
```diff
@@ -211,8 +211,20 @@ public static object GetDefaultValue(this Type type)
...
+
+        public static IEnumerable<TypeInfo> GetLoadableDefinedTypes(this Assembly assembly)
+        {
+            try
+            {
+                return assembly.DefinedTypes;
+            }
+            catch (ReflectionTypeLoadException ex)
+            {
+                return ex.Types.Select(IntrospectionExtensions.GetTypeInfo);
```
Comment (natemcmaster) [inline src/Shared/SharedTypeExtensions.cs]: Can cause NREs? ReflectionTypeLoadException.Types array can contain null values.

**Comment by natemcmaster** [comment]
Assuming the NRE isn't an issue, :shipit:

File: src/Shared/SharedTypeExtensions.cs (hunk 1 above)
Comment (bricelam) [inline src/Shared/SharedTypeExtensions.cs]: Good catch.

Comment (bricelam) [inline src/Shared/SharedTypeExtensions.cs]: I should've copied [it from EF6](https://github.com/bricelam/EntityFramework6/blob/master/src/Common/AssemblyExtensions.cs#L40). :wink:

**Pull request merged**